File: api-gateway/internal/model/analytics.go
package model

// AnalyticsSummary is the inner object for GET /api/analytics/summary (field "summary").
type AnalyticsSummary struct {
	TotalApplications    int                 `json:"total_applications"`
	ApplicationsByStatus map[string]int      `json:"applications_by_status"`
	Emails               AnalyticsEmailStats `json:"emails"`
	Rates                AnalyticsRates      `json:"rates"`
}

// AnalyticsEmailStats counts emails linked to the user's applications, by email.status.
type AnalyticsEmailStats struct {
	Sent      int `json:"sent"`
	Scheduled int `json:"scheduled"`
	Failed    int `json:"failed"`
	Draft     int `json:"draft"`
}

// AnalyticsRates are derived metrics; see service layer comments for exact formulas.
type AnalyticsRates struct {
	ReplyRate     float64 `json:"reply_rate"`
	InterviewRate float64 `json:"interview_rate"`
}

File: frontend/src/types/analytics.ts
import type { ApplicationStatus } from "./application";

/** Email aggregate counts from GET /api/analytics/summary (see docs/IMPLEMENTATION-GUIDE.md Appendix B). */
export interface AnalyticsEmailsSummary {
  sent: number;
  scheduled: number;
  failed: number;
  draft: number;
}

/**
 * Optional rates from the API; meaning is defined server-side (denominator in backend comments).
 */
export interface AnalyticsRatesSummary {
  reply_rate?: number;
  interview_rate?: number;
}

/**
 * Full analytics payload — `applications_by_status` should include every LOV key (zeros where empty).
 */
export interface AnalyticsSummary {
  total_applications: number;
  applications_by_status: Record<ApplicationStatus, number>;
  emails: AnalyticsEmailsSummary;
  rates?: AnalyticsRatesSummary;
}

export interface AnalyticsSummaryResponse {
  summary: AnalyticsSummary;
}
